staging/src/k8s.io/apiserver/pkg/util/flowcontrol/reqmgmt-config.go

```diff
@@ -0,0 +1,289 @@
+				klog.V(3).Infof("Priority level %s became undesired", plName)
+				plState.emptyHandler = &emptyRelay{reqMgmt: reqMgmt}
+				newlyQuiescent = append(newlyQuiescent, plState)
+			}
+			newRMState.priorityLevelStates[plName] = plState
+			shareSum += float64(plState.config.AssuredConcurrencyShares)
+		} else {
+			// keeps exempt or global-default
+			newRMState.priorityLevelStates[plName] = plState
+		}
+	}
+	for plName, plState := range newRMState.priorityLevelStates {
+		if plState.config.Exempt {
+			klog.V(5).Infof("Using exempt priority level %s: quiescent=%v", plName, plState.emptyHandler != nil)
+			continue
+		}
+		plState.concurrencyLimit = int(math.Ceil(float64(reqMgmt.serverConcurrencyLimit) * float64(plState.config.AssuredConcurrencyShares) / shareSum))
+		reqMgmt.syncPriorityLevelConcurrencyStatus(plByName[plName], plState.concurrencyLimit)
+		if plState.queues == nil {
+			klog.V(5).Infof("Introducing priority level %s: config=%#+v, concurrencyLimit=%d, quiescent=%v (shares=%v, shareSum=%v)", plName, plState.config, plState.concurrencyLimit, plState.emptyHandler != nil, plState.config.AssuredConcurrencyShares, shareSum)
+			plState.queues = reqMgmt.queueSetFactory.NewQueueSet(plState.concurrencyLimit, int(plState.config.Queues), int(plState.config.QueueLengthLimit), reqMgmt.requestWaitLimit)
+		} else {
+			klog.V(5).Infof("Retaining priority level %s: config=%#+v, concurrencyLimit=%d, quiescent=%v (shares=%v, shareSum=%v)", plName, plState.config, plState.concurrencyLimit, plState.emptyHandler != nil, plState.config.AssuredConcurrencyShares, shareSum)
+			plState.queues.SetConfiguration(plState.concurrencyLimit, int(plState.config.Queues), int(plState.config.QueueLengthLimit), reqMgmt.requestWaitLimit)
+		}
+	}
+	reqMgmt.curState.Store(newRMState)
+	klog.V(5).Infof("Switched to new RequestManagementState")
+	// We do the following only after updating curState to guarantee
+	// that if Wait returns `quiescent==true` then a fresh load from
+	// curState will yield an requestManagementState that is at least
+	// as up-to-date as the data here.
+	for _, plState := range newlyQuiescent {
+		plState.queues.Quiesce(plState.emptyHandler)
+	}
+}
+
+func (reqMgmt *requestManagementSystem) syncFlowSchemaStatus(fs *rmtypesv1a1.FlowSchema, isDangling bool) {
+	danglingCondition := rmtypesv1a1.GetFlowSchemaConditionByType(fs, rmtypesv1a1.FlowSchemaConditionDangling)
+	if danglingCondition == nil {
+		danglingCondition = &rmtypesv1a1.FlowSchemaCondition{
+			Type: rmtypesv1a1.FlowSchemaConditionDangling,
+		}
+	}
+
+	switch {
+	case isDangling && danglingCondition.Status != corev1.ConditionTrue:
+		danglingCondition.Status = corev1.ConditionTrue
+		danglingCondition.LastTransitionTime = metav1.Now()
+	case !isDangling && danglingCondition.Status != corev1.ConditionFalse:
+		danglingCondition.Status = corev1.ConditionFalse
+		danglingCondition.LastTransitionTime = metav1.Now()
+	}
+
+	rmtypesv1a1.SetFlowSchemaCondition(fs, *danglingCondition)
```

Comment:
It's worth attempting this only if it is a change, and emitting a log message on success as well as failure.